src/flexmeasures_client/client.py

```diff
@@ -675,6 +684,15 @@ async def update_asset(self, asset_id: int, updates: dict) -> dict:
         uri = f"assets/{asset_id}"
         if updates.get("attributes"):
             updates["attributes"] = json.dumps(updates["attributes"])
+        if updates.get("flex_context"):
+            updates["flex_context"] = json.dumps(updates["flex_context"])
+        if updates.get("sensors_to_show"):
+            updates["sensors_to_show"] = json.dumps(updates["sensors_to_show"])
```

Comment:
Wouldn't these fail to update in case of updating to an empty dict or list?

Reply:
Correct, the if-condition should be clearer, to check for `None`. I guess that is what you meant.
Also holds for the existing update for `attributes`.